lib/internal/timers.js

```diff
@@ -429,6 +432,20 @@ function setPosition(node, pos) {
   node.priorityQueuePosition = pos;
 }
 
+function removeAllStores (timer) {
```

Comment:
What about moving this function into async_hooks to avoid the need to expose `getActiveStores()` there?
An alternative would be to use a single object on the resource which holds all stores. This would reduce the number of properties added on resource objects - and avoid the need of a for loop to clear it.

Reply:
A helper that receives a resource and clears all the store data off it would be better than exposing the stores list as we do now. As it is presently, one could get and retain access to a store in a way unintentional from the perspective of the store owner, which is a bit dangerous from a memory perspective.